### ISSUE TITLE
ResponseMetadata attribute being added to SageMaker TypeDefs where it doesn't belong

### ISSUE THREAD

**Issue by jasonnance** [vemel/mypy_boto3_builder#91]
**Describe the bug**
Some TypeDefs related to SageMaker have spurious required ResponseMetadata attributes added to them.  I can see in the AWS API docs that the attribute doesn't belong there.  Example: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ProcessingS3Output.html

**To Reproduce**
1. Install 'boto3-stubs[sagemaker]'
2.  Run mypy on the following code sample

```python
from mypy_boto3_sagemaker.type_defs import ProcessingS3OutputTypeDef

d = ProcessingS3OutputTypeDef(
    S3Uri="s3://abc",
    LocalPath="/mydir",
    S3UploadMode="EndOfJob",
)
```
3. Actual mypy output

```
arc_ml_platform/type_test.py:3: error: Missing key 'ResponseMetadata' for TypedDict "ProcessingS3OutputTypeDef"
Found 1 error in 1 file (checked 1 source file)
```

**Expected mypy-output**

```
Success: no issues found in 1 source file
```

**Additional context**
OS: Linux (ubuntu)
Install method: PyPI via pip
Versions:
```
boto3                1.17.100
boto3-stubs          1.17.100
mypy-boto3-ecr       1.17.100
mypy-boto3-sagemaker 1.17.100
```

I see the line here adds ResponseMetadata for any types with "Output" at the end of the name: https://github.com/vemel/mypy_boto3_builder/blob/de115f93b8bcec6a0d290d5b11f10be454f3f6f7/mypy_boto3_builder/parsers/shape_parser.py#L339  This appears to be a little too lax, since several SageMaker objects end with "Output" but don't contain this attribute.  I'm not sure how to systematically identify cases like this where it isn't appropriate, but there are several SageMaker TypeDefs with names like this associated with the different job types (Processing, Training, Transform, etc).

**Comment by vemel** [comment]
Thank you! I will check if there is a flag for output shapes.

**Comment by vemel** [comment]
I just pushed a fix. I will try to release a new version tomorrow. So, instead of checking shape name, I add `ResponseMetadata` to method return type shape if it is a Mapping.

**Comment by vemel** [comment]
I have just released a new version with the bug fix. New `boto3-stubs` that includes the fix will be released tomorrow.

Please let me know if `boto3-stubs 1.17.102` works as it should for you.

**Comment by jasonnance** [comment]
Awesome, thanks so much for the quick response!  I'll check it out.  I don't see `boto3-stubs 1.17.102` on PyPI yet -- is there another distribution channel?

**Comment by vemel** [comment]
No, it will be built once `boto3 1.17.102` is out. Should be released in 5-6 hours. I will probably release `boto3-stubs 1.17.101.post1` just to cross-check if everything works as it should.

**Comment by vemel** [comment]
Version `1.17.101.post1` released with the bug fix. You can already test it if you have time.

Once again, let me know how it works for you.

**Comment by jasonnance** [comment]
Looks like that fixed it -- thanks again!

The only other thing I noticed is at least one of the typedef names changed: under SageMaker, `CreateProcessingJobResponseTypeDef` is now `CreateProcessingJobResponseResponseTypeDef`.  That's unrelated to this issue, though -- just an FYI, not sure if it was intended.

**Comment by vemel** [comment]
it is intended. I add `ResponseTypeDef` postfix to all shapes with `ResponseMetadata`. Otherwise, they conflict with the same shape when it is not used as a response.
